code/game/objects/items/bodybag.dm

```diff
@@ -84,6 +84,33 @@
 	else
 		icon_state = icon_opened
 
+/obj/structure/closet/body_bag/take_contents()
+	..()
+	var/mob/living/L = usr
+	for(L in src)
+		if (L.resting == FALSE) //if you are standing up and the body bag closes...
+			L.lay_down() //rest
+
+/obj/structure/closet/body_bag/close()
+	..()
+	if(usr)
```

Comment:
usr

Reply:
This check also just doesn't work properly. It'll only work if the person closes it on themselves, not if someone else closes it on them.